tests/_mocks/abstract_presentation.php

```diff
@@ -0,0 +1,15 @@
+<?php
+namespace Yoast\WP\SEO\Tests\Mocks;
+
+class Abstract_Presentation extends \Yoast\WP\SEO\Presentations\Abstract_Presentation {
+	/**
+	 * @inheritDoc
+	 */
+	public function is_prototype() {
+		return parent::is_prototype();
+	}
+
+	public function generate_some_mock_data() {
```

Comment:
Docs missing

Reply:
It seems this function isn't called anywhere?

Reply:
> It seems this function isn't called anywhere?

It is called, but not directly. 

In `abstract-presentation` we have this code that generates the method call on the fly, based on the property that is used.
https://github.com/Yoast/wordpress-seo/blob/3964fd5bd2e82ab9309b8e2bdf5fd7e084004df7/src/presentations/abstract-presentation.php#L63-L73

It goes something like this
 1. `$presentation->some_mock_data` is used somewhere in the code.
 2. PHP notices that this property is not accessible, in this case it does not exist yet.
 2. PHP translates the call to `$presentation->__get('some_mock_data`)`.
 3. The above method gets called.
 4. The method calls the appropriate generate method, in this case `generate_some_mock_data`, sets its results on the appropriate property and returns it.
 5. All subsequent uses of `$presentation->some_mock_data` uses the property, since it now exists since step 4.